app/src/main/java/com/github/se/cyrcle/ui/list/ListScreen.kt

```diff
@@ -262,67 +314,163 @@ fun FilterSection(
+                .height(120.dp)
+                .offset { IntOffset(offsetX.roundToInt(), 0) }
+                .pointerInput(Unit) {
+                  detectHorizontalDragGestures(
+                      onDragEnd = {
+                        if (offsetX > maxSwipeDistance.toPx() / 2) {
+                          // Swipe right action: Toggle pin status
+                          parkingViewModel.togglePinStatus(parking)
+                        } else if (offsetX < -maxSwipeDistance.toPx() / 2) {
+                          // Swipe left action: Add to favorites
+                          if (!isFavorite && userSignedIn.value) {
+                            userState?.let {
+                              userViewModel.addFavoriteParkingToSelectedUser(parking.uid)
+                              userViewModel.getSelectedUserFavoriteParking()
+                            }
+                          } else if (!userSignedIn.value) {
+                            Toast.makeText(
+                                    context, "Please sign in to add favorites", Toast.LENGTH_SHORT)
+                                .show()
+                          } else if (isFavorite) {
+                            Toast.makeText(
+                                    context, "Parking is already in favorites", Toast.LENGTH_SHORT)
+                                .show()
+                          }
+                        }
+                        offsetX = 0f
+                      }) { change, dragAmount ->
+                        change.consume()
+                        offsetX =
+                            (offsetX + dragAmount).coerceIn(
+                                -maxSwipeDistance.toPx(), maxSwipeDistance.toPx())
+                      }
+                }
+                .clickable(
+                    onClick = {
+                      parkingViewModel.selectParking(parking)
+                      navigationActions.navigateTo(Screen.PARKING_DETAILS)
+                    })
+                .testTag("SpotListItem"),
+        colors =
+            CardDefaults.cardColors(
+                containerColor = MaterialTheme.colorScheme.surface.copy(alpha = 1f)),
+        shape = MaterialTheme.shapes.medium) {
+          Box(modifier = Modifier.fillMaxSize()) {
+            Column(modifier = Modifier.fillMaxSize().padding(16.dp).testTag("SpotCardContent")) {
+              Row(
+                  modifier = Modifier.fillMaxWidth(),
+                  verticalAlignment = Alignment.CenterVertically,
+                  horizontalArrangement = Arrangement.SpaceBetween) {
+                    Text(
+                        text =
+                            parking.optName?.let { if (it.length > 35) it.take(32) + "..." else it }
+                                ?: stringResource(R.string.default_parking_name),
+                        style = MaterialTheme.typography.bodyLarge,
+                        color = MaterialTheme.colorScheme.onSurface,
```

Comment:
Please can you remove the color argument that André's PR carefully removed? Also valid for line 345